src/adapter/src/catalog/builtin.rs

```diff
@@ -162,16 +163,28 @@ impl Fingerprint for BuiltinFunc {
 impl<T: TypeReference> Fingerprint for &Builtin<T> {
     fn fingerprint(&self) -> u64 {
         match self {
-            Builtin::Log(log) => log.fingerprint(),
-            Builtin::Table(table) => table.fingerprint(),
-            Builtin::View(view) => view.fingerprint(),
+            Builtin::Log(log) => log.variant.desc().fingerprint(),
+            Builtin::Table(table) => table.desc.fingerprint(),
+            Builtin::View(view) => view.sql.hashed(),
             Builtin::Type(typ) => typ.fingerprint(),
             Builtin::Func(func) => func.fingerprint(),
-            Builtin::StorageCollection(coll) => coll.fingerprint(),
+            Builtin::StorageCollection(coll) => coll.desc.fingerprint(),
         }
     }
 }
 
+impl Fingerprint for RelationDesc {
+    fn fingerprint(&self) -> u64 {
+        self.typ().fingerprint()
+    }
+}
+
+impl Fingerprint for RelationType {
```

Comment:
This is still probably even more aggressive than we ultimately want. Somewhere nested in `RelationType` is `ScalarType`. When a new type is added, the derived hash implementation changes and will change the fingerprint for all objects that use this impl.